cpp/ConnectionPool.cpp

```diff
@@ -0,0 +1,330 @@
+      writeQueue.erase(writeQueue.begin());
+    } else {
+      // No items in the queue, clear the context
+      writeConnection.currentLockId = EMPTY_LOCK_ID;
+    }
+  } else {
+    // Check if it's a read connection
+    for (int i = 0; i < maxReads; i++) {
+      if (readConnections[i].currentLockId == contextId) {
+        if (readQueue.size() > 0) {
+          // There are items in the queue, activate the next one
+          activateContext(&readConnections[i], readQueue[0]);
+          readQueue.erase(readQueue.begin());
+        } else {
+          // No items in the queue, clear the context
+          readConnections[i].currentLockId = EMPTY_LOCK_ID;
+        }
+        return;
+      }
+    }
+  }
+}
+
+void ConnectionPool::closeAll() {
+  sqlite3_close_v2(writeConnection.connection);
+  for (int i = 0; i < maxReads; i++) {
+    sqlite3_close_v2(readConnections[i].connection);
+  }
+}
+
+SQLiteOPResult ConnectionPool::attachDatabase(std::string const dbFileName,
+                                              std::string const docPath,
+                                              std::string const alias) {
+
+  /**
+   * There is no need to check if mainDBName is opened because
+   * sqliteExecuteLiteral will do that.
+   * */
+  string dbPath = get_db_path(dbFileName, docPath);
+  string statement = "ATTACH DATABASE '" + dbPath + "' AS " + alias;
+
+  auto dbConnections = getAllConnections();
+
+  for (auto &connection : dbConnections) {
+    SequelLiteralUpdateResult result =
+        sqliteExecuteLiteralWithDB(connection, statement);
+    if (result.type == SQLiteError) {
+      // Revert change on any successful connections
+      detachDatabase(alias);
+      return SQLiteOPResult{
+          .type = SQLiteError,
+          .errorMessage = dbName + " was unable to attach another database: " +
+                          string(result.message),
+      };
+    }
+  }
+
+  return SQLiteOPResult{
+      .type = SQLiteOk,
+  };
+}
+
+SQLiteOPResult ConnectionPool::detachDatabase(std::string const alias) {
+  /**
+   * There is no need to check if mainDBName is opened because
+   * sqliteExecuteLiteral will do that.
+   * */
+  string statement = "DETACH DATABASE " + alias;
+  auto dbConnections = getAllConnections();
+
+  for (auto &connection : dbConnections) {
+    SequelLiteralUpdateResult result =
+        sqliteExecuteLiteralWithDB(connection, statement);
+    if (result.type == SQLiteError) {
+      return SQLiteOPResult{
+          .type = SQLiteError,
+          .errorMessage = dbName + " was unable to attach another database: " +
+                          string(result.message),
+      };
+    }
+  }
+  return SQLiteOPResult{
+      .type = SQLiteOk,
+  };
+}
+
+SequelBatchOperationResult
+ConnectionPool::importSQLFile(std::string fileLocation) {
+  std::string line;
+  std::ifstream sqFile(fileLocation);
+
+  if (sqFile.is_open()) {
+    sqlite3 *connection = writeConnection.connection;
```

Comment:
This bypasses the locking logic - could result in concurrency issues.

Reply:
This method is only intended to be used via the connection wrapped in a `writeLock` https://github.com/journeyapps/react-native-quick-sqlite/blob/fd08cef2dac9fa37f97859a2cff9611544ac9e66/src/setup-open.ts#L222. While it may be possible to call the method via the global module, that should not be recommend as using the module directly can bypass certain JS checks in some circumstances.